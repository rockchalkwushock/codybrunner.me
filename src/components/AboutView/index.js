/* eslint-disable react/no-danger */
import React from 'react'

import { Content, Header, Post, Share } from '../Post'
import SEO from '../SEO'

/**
 * @fileOverview
 * About Page View
 *
 * Accepts data for markdown file /content/posts/about.md.
 * Will present the updated 'fileTime', no date in in frontmatter.
 */

const AboutView = ({ content, meta, modTime }) => {
  const { fields, frontmatter, html, timeToRead, wordCount } = content
  // REVIEW: trailing slash, problematic? Plugin for handling this.
  const aboutUrl = `${meta.siteUrl}${fields.slug}`
  return (
    <Post draft={frontmatter.draft}>
      <SEO site={meta} />
      <Header
        about
        date={modTime}
        time={timeToRead}
        title={frontmatter.title}
        words={wordCount.words}
      />
      <Content dangerouslySetInnerHTML={{ __html: html }} />
      <Share title={frontmatter.title} url={aboutUrl} />
    </Post>
  )
}

export default AboutView
