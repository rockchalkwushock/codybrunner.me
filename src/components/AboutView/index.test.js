import React from 'react'
import { shallow } from 'enzyme'

import AboutView from './'

const { siteMetadata } = require('../../../gatsby-config')

const content = {
  fields: {
    slug: '/turdsspace/'
  },
  frontmatter: {
    draft: false,
    title: 'All About JS'
  },
  html: 'hello world',
  timeToRead: 10
}

describe('Component: <AboutView />', () => {
  let wrapper
  beforeEach(() => {
    wrapper = shallow(
      <AboutView content={content} meta={siteMetadata} modTime="02-03-99" />
    )
  })

  test('should render correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
  test('should have matching content on html', () => {
    expect(
      wrapper.find('Content').prop('dangerouslySetInnerHTML').__html
    ).toEqual('hello world')
  })
})
