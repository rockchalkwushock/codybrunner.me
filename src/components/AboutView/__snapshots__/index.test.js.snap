// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <AboutView /> should render correctly 1`] = `
<Post
  draft={false}
>
  <SEO
    site={
      Object {
        "aboutSnippet": "Cody Brunner is a full-stack JavaScript developer & Navy Veteran residing in Wichita, Kansas. Cody primarily works with Node, React, & GraphQL. When not writing code he loves to go hiking, play with his dog & niece, and watchhis Jayhawks win!",
        "author": "Cody Brunner",
        "business": Array [
          Object {
            "className": "fas fa-envelope",
            "href": "mailto:rockchalkwushock@icloud.com",
            "label": "Email",
          },
          Object {
            "className": "fab fa-telegram-plane",
            "href": "https://t.me/rockchalkwushock",
            "label": "Telegram",
          },
          Object {
            "className": "fas fa-file-pdf",
            "href": "https://www.dropbox.com/s/hokjljqc8iob7xd/Cody%20A%20Brunner%20-%20Web%20Developer%20Resume.pdf?dl=1",
            "label": "Resume",
          },
        ],
        "contacts": Array [
          Object {
            "className": "fab fa-github",
            "href": "https://github.com/rockchalkwushock",
            "label": "Github",
          },
          Object {
            "className": "fab fa-instagram",
            "href": "https://www.instagram.com/rockchalkwushock",
            "label": "Instagram",
          },
          Object {
            "className": "fab fa-linkedin",
            "href": "https://www.linkedin.com/in/cody-brunner-324930158",
            "label": "LinkedIn",
          },
          Object {
            "className": "fas fa-rss",
            "href": "rss.xml",
            "label": "RSS Feed",
          },
          Object {
            "className": "fab fa-twitter",
            "href": "https://twitter.com/RockChalkDev",
            "label": "Twitter",
          },
          Object {
            "className": "fab fa-youtube",
            "href": "https://www.youtube.com/channel/UCZgBTMhX7jZTkbm7Fpv2bWw",
            "label": "Youtube",
          },
        ],
        "copyright": "Â© 2017-2018 Cody Brunner",
        "description": "Cody Brunner is a Full-Stack JavaScript Developer living in Wichita, Kansas",
        "disqusShortname": "codybrunner-rocks",
        "employment": "Are you interested in contacting Cody for employment opportunities? Feel free to reach out to him via email, Telegram, or go ahead and checkout his resume hosted on DropBox.",
        "googleVerify": "TODO",
        "jobTitle": "Full-Stack JavaScript Developer",
        "keywords": "Cody Brunner, Full-Stack JavaScript Developer, Wichita, web-development, tech blog, Node.js, React.js, Apollo/GraphQL",
        "lang": "en_US",
        "links": Object {
          "creativeCommons": Object {
            "href": "https://creativecommons.org/licenses/by-nc-sa/4.0/",
            "text": "All Rights Reserved.",
          },
          "gatsby": Object {
            "href": "https://gatsbyjs.org",
            "text": "Powered by Gatsby 1.9",
          },
          "now": Object {
            "href": "https://zeit.co/now",
            "text": "Hosted on Now",
          },
          "src": Object {
            "href": "https://github.com/rockchalkwushock/codybrunner.rocks",
          },
          "styled": Object {
            "href": "https://www.styled-components.com",
            "text": "Styled with Styled-Components 3.1",
          },
        },
        "menu": Array [
          Object {
            "href": "/",
            "id": 1,
            "text": "Home",
          },
          Object {
            "href": "/about",
            "id": 2,
            "text": "About",
          },
          Object {
            "href": "/posts",
            "id": 3,
            "text": "Blog",
          },
          Object {
            "href": "/tags",
            "id": 4,
            "text": "Tags",
          },
        ],
        "siteUrl": "/",
        "title": "Cody Brunner - Full-Stack JavaScript Developer",
        "twitter": "@RockChalkDev",
      }
    }
  />
  <PostHeader
    about={true}
    date="02-03-99"
    time={10}
    title="All About JS"
  />
  <Content
    dangerouslySetInnerHTML={
      Object {
        "__html": "hello world",
      }
    }
  />
  <Share
    title="All About JS"
    url="//turdsspace/"
  />
</Post>
`;
